apps/src/applab/applab.js

```diff
@@ -624,18 +624,25 @@ Applab.init = function (config) {
 
   Applab.reactMountPoint_ = document.getElementById(config.containerId);
 
-  Applab.render();
-
-  //Scale old-sized apps to fit the new sized display. Old height - 480.
-  if ($(".screen").height() === 480) {
-    const ratio = 450 / 480;
-    if (studioApp().share) { //share and embed pages
-      $("#divApplab").css('transform', 'scale(' + ratio + ', ' + ratio + ')');
-      $(".small-footer-base").css('transform', 'scale('+ ratio + ', 1)');
-    } else { //includes the frame on the edit page
-      $("#phoneFrameWrapper").css('transform', 'scale(' + ratio + ', ' + ratio + ')');
+  const loader = studioApp().loadLibraries_(level.helperLibraries).then(() => {
+    Applab.render();
+
+    //Scale old-sized apps to fit the new sized display. Old height - 480.
+    if ($(".screen").height() === 480) {
+      const ratio = 450 / 480;
+      if (studioApp().share) { //share and embed pages
+        $("#divApplab").css('transform', 'scale(' + ratio + ', ' + ratio + ')');
+        $(".small-footer-base").css('transform', 'scale('+ ratio + ', 1)');
+      } else { //includes the frame on the edit page
+        $("#phoneFrameWrapper").css('transform', 'scale(' + ratio + ', ' + ratio + ')');
+      }
     }
+  });
+
+  if (IN_UNIT_TEST) {
+    return loader.catch(() => {});
```

Comment:
What's this for?

Reply:
I'm following the pattern in commit https://github.com/code-dot-org/code-dot-org/commit/914ee28e520c1c199864d27d09d56a54c73d700e.  When I did this in Game Lab, I was getting random karma:integration test failures when the pending promise would reject after the test finished, when another test was running.

Reply:
I remember Brent doing some work to deal with state leaking across test runs... but that's all I remember :-/ I'll see if I can dig up any alternate solution.

please correct me if I'm wrong, but it seems like the downside here is that a legit error inside the call to `loadLibraries_` would not be caught by this test. on the plus side... the tests are passing, and still succeed in testing everything outside of this block.

Reply:
Oh man, debugging integration test issues... I'd like to say any tests hitting this code should stub these requests, but in practice that's not going to happen anytime soon.  This seems like a reasonable bit of tech debt to trade for getting this feature up and running now.  A comment about what failure mode we're avoiding would be nice though.